### PR TITLE
UIHelperExtension, UITable

### PR DESCRIPTION
**Pull request by xlf1024**

### CODE REVIEW

File: KianCommons/UI/Table/UITableRow.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                var cell = AddUIComponent<UITableCellOuter>();
+                cell.columnIndex = columnIndex;
+                cell.rowIndex = rowIndex;
+                return cell;
+            });
+        }
+
+        internal void Shrink(int columnCount) {
+            cells = cells.Shrink(columnCount, (cell, columnIndex) => {
+                RemoveUIComponent(cell);
```
Comment (kianzarrin) [inline KianCommons/UI/Table/UITableRow.cs]: [minor comment]
this line is redundant.  components are removed when they are destroyed.

File: KianCommons/UI/Table/UITable.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
+using ColossalFramework.UI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace KianCommons.UI.Table {
+    class UITable : UIPanel {
+        public UITableRow[] rows = new UITableRow[0];
```
Comment (kianzarrin) [inline KianCommons/UI/Table/UITable.cs]: public fiedls should be CamelCase
private fields should be camelCase_

Also do you really want to expose this field? maybe it should be private with property to access it.

File: KianCommons/UI/Table/UITable.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        private float[] columnWidths = new float[0];
+        public int RowCount {
+            get => rows.Length;
+        }
```
Comment (kianzarrin) [inline KianCommons/UI/Table/UITable.cs]: [minor comment]
you can also write
public int RowCount => rows.Length

File: KianCommons/UI/Table/UITableRow.cs (hunk 1 above)
Comment (xlf1024) [reply KianCommons/UI/Table/UITableRow.cs]: removed.

File: KianCommons/UI/Table/UITable.cs (hunk 2 above)
Comment (xlf1024) [reply KianCommons/UI/Table/UITable.cs]: changed, now private. Though I might need to add a getter later for styling.

File: KianCommons/UI/Table/UITable.cs (hunk 3 above)
Comment (xlf1024) [reply KianCommons/UI/Table/UITable.cs]: done

File: KianCommons/UI/Table/UITable.cs (hunk 2 above)
Comment (xlf1024) [reply KianCommons/UI/Table/UITable.cs]: added the trailing underscore; mistook that for a typo first.

File: KianCommons/UI/Table/UITableCellInner.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            atlas = TextureUtil.Ingame;
+            eventSizeChanged += (_, _) => {
+                try { // the first size change occurs within UITableCellOuter::Activate, at which point UITableCellOuter.parent isn't a UITableRow yet.
+                    var _outerCell = parent as UITableCellOuter;
+                    var _row = _outerCell.parent as UITableRow;
+                    var _table = _row.parent as UITable;
+                    _table.ResizeColumn(_outerCell.ColumnIndex);
+                }catch(NullReferenceException _) {
+
+                }
```
Comment (xlf1024) [inline KianCommons/UI/Table/UITableCellInner.cs]: I wonder if there's a better way to handle this?

Comment (xlf1024) [reply KianCommons/UI/Table/UITableCellInner.cs]: (just referring to the use of try...catch)

Comment (xlf1024) [reply KianCommons/UI/Table/UITableCellInner.cs]: resolved; replaced with `?.`

**Pull request merged**